src/test/java/com/example/amazoncom/stepsDefs/HomePageSteps.java

```diff
@@ -0,0 +1,87 @@
+package com.example.amazoncom.stepsDefs;
+
+import com.example.amazoncom.pageObjects.AmazonMainPage;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.assertj.core.api.Assertions;
+public class HomePageSteps {
+
+  AmazonMainPage mainPage = new AmazonMainPage();
+
+  @Given("Amazon page is opened")
+  public void openAmazonHomePage() {
+    mainPage.openHomePage();
+    mainPage.acceptCookies.click();
+  }
+
+  @When("Type {string} into the search field")
+  public void typeValueIntoTheSearchField(String searchValue) {
+    mainPage.searchField.sendKeys(searchValue);
+  }
+
+  @When("Click on the search button")
+  public void clickSearchButton() {
+    mainPage.searchButton.click();
+  }
+
+  @Then("Check search result contains {string}")
+  public void checkSearchResultContainsValue(String searchValue) {
+    // Assertions.assertEquals(searchValue, mainPage.getSearchFieldValue());
+    Assertions.assertThat(mainPage.getSearchFieldValue())
+        .as("Failure message: search button doesn't work")
+        .containsIgnoringCase(searchValue);
+  }
+
+
+  //form to login
+  @When("Click on the sign up button")
+  public void clickSignUpButton() {
+    mainPage.signUpButton.click();
+  }
+
+  @Then("Check page contain {string}")
```

Comment:
containS

Reply:
and your test is about "checkTitleLogin" mayme to describe this in the test step description? like "check login form is displayed"